pydantic/mypy.py

```diff
@@ -0,0 +1,624 @@
+        expr = stmt.rvalue
+        if not isinstance(expr, TempNode):
+
+            if (
+                isinstance(expr, CallExpr)
+                and isinstance(expr.callee, RefExpr)
+                and expr.callee.fullname == FIELD_FULLNAME
+            ):
+                arg_is_ellipsis = len(expr.args) > 0 and type(expr.args[0]) is EllipsisExpr
+            else:
+                arg_is_ellipsis = isinstance(expr, EllipsisExpr)
+            return arg_is_ellipsis
+        # I believe TempNode means annotation-only
+        value_type = cls.info[lhs.name].type
+        if type(value_type) is UnionType:
+            item_types = [type(item) for item in value_type.items]  # type: ignore
+            if NoneType in item_types:
+                return False
+        return True
+
+    @staticmethod
+    def get_alias_info(stmt: AssignmentStmt) -> Tuple[Optional[str], bool]:
+        """
+        Returns (alias, has_dynamic_alias)
+        """
+        expr = stmt.rvalue
+        if isinstance(expr, TempNode):
+            # I believe TempNode means annotation-only
+            return None, False
+
+        if not (
+            isinstance(expr, CallExpr) and isinstance(expr.callee, RefExpr) and expr.callee.fullname == FIELD_FULLNAME
+        ):
+            # Assigned value is not a call to pydantic.fields.Field
+            return None, False
+
+        for i, arg_name in enumerate(expr.arg_names):
+            if arg_name != 'alias':
+                continue
+            arg = expr.args[i]
+            if isinstance(arg, StrExpr):
+                return arg.value, False
+            else:
+                return None, True
+        return None, False
+
+    def get_field_arguments(
+        self, fields: List['PydanticModelField'], typed: bool, force_all_optional: bool, use_alias: bool
+    ) -> List[Argument]:
+        info = self._ctx.cls.info
+        arguments = [
+            field.to_argument(info, typed=typed, force_optional=force_all_optional, use_alias=use_alias)
+            for field in fields
+            if not (use_alias and field.has_dynamic_alias)
+        ]
+        return arguments
+
+    def should_init_allow_extra(self, fields: List['PydanticModelField'], config: 'ModelConfigData') -> bool:
+        if not config.allow_population_by_field_name:
+            if self.is_dynamic_alias_present(fields, bool(config.has_alias_generator)):
+                return True
+        if config.forbid_extra:
+            return False
+        return self.plugin_config.init_allow_extra
+
+    @staticmethod
+    def is_dynamic_alias_present(fields: List['PydanticModelField'], has_alias_generator: bool) -> bool:
+        for field in fields:
+            if field.has_dynamic_alias:
+                return True
+        if has_alias_generator:
+            for field in fields:
+                if field.alias is None:
+                    return True
+        return False
+
+
+class PydanticModelField:
```

Comment:
I wanted to do that, but it would break compatibility with 3.6 without the dataclasses backport. Let me know if you would want that despite the restriction it would add

Reply:
But the backport of dataclasses is a required dependency.

Unless there's some reasonable way to use this plugin without installing pydantic (and therefore dataclasses), I think it's fine to use dataclasses.

Up to you though.

Reply:
I see, I didn't realize that. Then yeah I think dataclasses is better for sure.